org-code-javabuilder/lib/src/main/java/org/code/neighborhood/Painter.java

```
@@ -0,0 +1,146 @@
+package org.code.neighborhood;
+
+public class Painter {
+  private static int lastId = 0;
+  private int xLocation;
+  private int yLocation;
+  private Direction direction;
+  private int remainingPaint;
+  private Grid grid;
+  private String id;
+
+  public Painter(int x, int y, String direction, int paint) {
+    this.xLocation = x;
+    this.yLocation = y;
+    this.direction = Direction.fromString(direction);
+    if (this.direction == null) {
+      throw new UnsupportedOperationException("Invalid direction given to painter");
+    }
+    this.remainingPaint = paint;
+    this.grid = World.getInstance().getGrid();
+    this.id = "painter-" + lastId++;
+  }
+
+  // Turn the painter one compass direction left (i.e. North -> West)
+  public void turnLeft() {
+    this.direction = this.direction.turnLeft();
+  }
+
+  // Move the painter one square forward in the direction the painter
+  // is facing
+  public void move() {
+    if (this.isValidMovement(this.direction)) {
+      if (this.direction.isNorth()) {
+        this.yLocation++;
+      } else if (this.direction.isSouth()) {
+        this.yLocation--;
+      } else if (this.direction.isEast()) {
+        this.xLocation++;
+      } else {
+        this.xLocation--;
+      }
+    } else {
+      System.out.println("You can't go that way");
```

We should throw an exception here.